src/Parser/Mage/Frost/Modules/Features/ThermalVoid.js

```diff
@@ -14,29 +14,38 @@ class ThermalVoid extends Module {
 	}
 
   casts = 0;
+  buffApplied = 0;
+  extraUptime = 0;
 
   on_initialized() {
 	   this.active = this.combatants.selected.hasTalent(SPELLS.THERMAL_VOID_TALENT.id);
   }
 
   on_toPlayer_applybuff(event) {
-    const spellId = event.ability.guid;
-    if(spellId === SPELLS.ICY_VEINS.id && event.prepull) { // catch prepull cast by buff being present on pull
+    if(event.ability.guid === SPELLS.ICY_VEINS.id && event.prepull) { // catch prepull cast by buff being present on pull
       this.casts += 1;
+      this.buffApplied = event.timestamp;
     }
   }
 
   on_byPlayer_cast(event) {
-    const spellId = event.ability.guid;
-    if (spellId !== SPELLS.ICY_VEINS.id) {
+    if (event.ability.guid !== SPELLS.ICY_VEINS.id) {
       return;
     }
     this.casts += 1;
+    this.buffApplied = event.timestamp;
```

Comment:
Rather than having one case for precasts with on_byPlayer_applybuff && prepull, and another for regular casts with on_byPlayer_cast, you could and probably should just catch both with on_byPlayer_applybuff.

Reply:
I think you were the one that added the prepull stuff, so im not quite sure how that prepull event works

Reply:
I may have added the extra handling, yeah. I'll fix it myself later. Nothing will work wrong, just noticing a cleaner solution.

Reply:
Yea i have no problem fixing it, just not sure how to combine them and make sure the pre pull works .... plus i dont know if i have a log with a prepull Icy Veins in it, so i dont know that i could even test it